akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -38,62 +43,75 @@ private[http] object OutgoingConnectionBlueprint {
                  +-------------------------------------->|  Merge   |                       |
                  | Termination Backchannel          |    +----------+                       |  TCP-
                  |                                  |                                       |  level
-                 |                                  | Method                                |  client
-                 |                +------------+    | Bypass                                |  flow
+                 |                                  | BypassData                            |  client
+                 |                +------------+    |                                       |  flow
     responseOut  |  responsePrep  |  Response  |<---+                                       |
     <------------+----------------|  Parsing   |                                            |
                                   |  Merge     |<------------------------------------------ V
                                   +------------+
   */
-  def apply(hostHeader: Host,
+  def apply(hostHeader: headers.Host,
             settings: ClientConnectionSettings,
             log: LoggingAdapter): Http.ClientLayer = {
     import settings._
 
-    // the initial header parser we initially use for every connection,
-    // will not be mutated, all "shared copy" parsers copy on first-write into the header cache
-    val rootParser = new HttpResponseParser(parserSettings, HttpHeaderParser(parserSettings) { info ⇒
-      if (parserSettings.illegalHeaderWarnings)
-        logParsingError(info withSummaryPrepended "Illegal response header", log, parserSettings.errorLoggingVerbosity)
-    })
+    val core = BidiFlow.fromGraph(GraphDSL.create() { implicit b ⇒
+      import GraphDSL.Implicits._
 
-    val requestRendererFactory = new HttpRequestRendererFactory(userAgentHeader, requestHeaderSizeHint, log)
+      val renderingContextCreation = b.add {
+        Flow[HttpRequest] map { request ⇒
+          val sendEntityTrigger =
+            request.headers collectFirst { case headers.Expect.`100-continue` ⇒ Promise[NotUsed]().future }
```

Comment:
`Promise[NotUsed]().future` will never complete, so any callbacks that are added to it will leak for as long as the Future is referenced. Is this intentional? (See: https://github.com/viktorklang/blog/blob/master/Futures-in-Scala-2.12-part-6.md#missing-utilities-unit--never )